pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CastTransformFunction.java

```diff
@@ -56,6 +56,10 @@ public void init(List<TransformFunction> arguments, Map<String, ColumnContext> c
     TransformFunction castFormatTransformFunction = arguments.get(1);
     if (castFormatTransformFunction instanceof LiteralTransformFunction) {
       String targetType = ((LiteralTransformFunction) castFormatTransformFunction).getStringLiteral().toUpperCase();
+      if ("OBJECT".equals(targetType)) {
```

Comment:
Hmm, what does `CAST to OBJECT` mean?

Reply:
This happens when scalar function is not explicitly registered for the return type.